resources/assets/coffee/react/beatmap-discussions-history/main.coffee

```diff
@@ -0,0 +1,150 @@
+###
+#    Copyright (c) ppy Pty Ltd <contact@ppy.sh>.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+import { Discussion } from '../beatmap-discussions/discussion'
+import { BeatmapsContext } from 'beatmap-discussions/beatmaps-context'
+import { DiscussionsContext } from 'beatmap-discussions/discussions-context'
+import * as React from 'react'
+import { a, div, img } from 'react-dom-factories'
+el = React.createElement
+
+export class Main extends React.PureComponent
+  constructor: (props) ->
+    super props
+
+    @cache = {}
+    @state = JSON.parse(props.container.dataset.discussionsState ? null)
+    @restoredState = @state?
+
+    if !@restoredState
+      @state =
+        discussions: props.discussions
+        users: props.users
+        relatedDiscussions: props.relatedDiscussions
+
+
+  componentDidMount: =>
+    $.subscribe 'beatmapsetDiscussions:update.discussionHistory', @discussionUpdate
+    $(document).on 'ajax:success.discussionHistory', '.js-beatmapset-discussion-update', @ujsDiscussionUpdate
+
+
+  componentWillUnmount: =>
+    $.unsubscribe '.discussionHistory'
+    $(window).off '.discussionHistory'
+
+    $(window).stop()
+
+
+  discussionUpdate: (_e, options) =>
```

Comment:
looks like sometimes editing someone else's post doesn't show correct "edited by" line

Reply:
any details on how to reproduce?

Reply:
I think it's when you're editing others' post and there's no other post/edit by you on the page when editing it.

Reply:
Hmm, that shouldn't matter though, 'cuz `discussionUpdate` iterates over the related_users and adds them... weird.